src/Core/src/PerformanceTracker/Extensions/PerformanceMauiAppBuilderExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System.Diagnostics.Metrics;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Maui.Hosting;
+
+namespace Microsoft.Maui.Performance
+{
+    /// <summary>
+    /// Extension methods to register and configure performance monitoring services for a MAUI application.
+    /// </summary>
+    internal static class PerformanceMauiAppBuilderExtensions
+    {
+        /// <summary>
+        /// Adds .NET MAUI performance monitoring services to the application's dependency injection container.
+        /// </summary>
+        /// <param name="builder">The <see cref="MauiAppBuilder"/> to which performance monitoring is being added.</param>
+        /// <returns>The same <see cref="MauiAppBuilder"/> instance.</returns>
+        public static MauiAppBuilder ConfigurePerformance(
+			this MauiAppBuilder builder)
+        {
+            // Check if the performance profiling feature is enabled.
+            if (RuntimeFeature.IsMetricsSupported)
+            {
+                // Register the Meter
+                var meter = new Meter("Microsoft.Maui");
+                builder.Services.AddSingleton(meter);
```

Comment:
You should probably use a different DI key for this. Using the `Meter` type as the key is too general. Some other library or even the app might want to add a Meter.

Can you used Keyed DI? Or even your own internal type that has a Meter if you can't use keyed?